src/Plugins/OrientationAnalysis/src/OrientationAnalysis/utilities/ImportH5Data.hpp

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "OrientationAnalysis/OrientationAnalysis_export.hpp"
+#include "OrientationAnalysis/Parameters/OEMEbsdScanSelectionParameter.h"
+
+#include "complex/DataStructure/DataStructure.hpp"
+#include "complex/Filter/IFilter.hpp"
+
+#include "EbsdLib/IO/BrukerNano/H5EspritReader.h"
+#include "EbsdLib/IO/TSL/H5OIMReader.h"
+
+namespace complex
+{
+struct ORIENTATIONANALYSIS_EXPORT ImportH5DataInputValues
+{
+  OEMEbsdScanSelectionParameter::ValueType SelectedScanNames;
+  bool ReadPatternData;
+  DataPath ImageGeometryPath;
+  DataPath CellAttributeMatrixPath;
+  DataPath CellEnsembleAttributeMatrixPath;
+};
+
+/**
+ * @class ImportH5Data
+ */
+
+template <class T>
+class ORIENTATIONANALYSIS_EXPORT ImportH5Data
+{
+public:
+  ImportH5Data(DataStructure& dataStructure, const IFilter::MessageHandler& mesgHandler, const std::atomic_bool& shouldCancel, const ImportH5DataInputValues* inputValues);
+  virtual ~ImportH5Data() noexcept;
+
+  ImportH5Data(const ImportH5Data&) = delete;
+  ImportH5Data(ImportH5Data&&) noexcept = delete;
+  ImportH5Data& operator=(const ImportH5Data&) = delete;
+  ImportH5Data& operator=(ImportH5Data&&) noexcept = delete;
+
+  Result<> execute();
```

Comment:
These need to be fully declared with the template also which means the code will have to come into the header. Or have all of the details on the inside of the class. macOS and Linux will not compile this code.

I think there are some `external` shenanigans that you can do to keep the implementation details out of the header.